### PR TITLE
MAINT - Moving negative path tests to separate files

### PR DESCRIPTION
**Pull request by brjeter**
#### What does this PR do (if it's not clear from the Title)? Please include any specification references that apply.
- Also changed names to say "SSO" instead of "Login"
#### Checklist:
- [ ] SAML Spec Table of Contents documentation updated

### CODE REVIEW

**Comment by brjeter** [comment]
build now

File: ctk/idp/src/main/kotlin/org/codice/compliance/web/sso/error/PostSSOErrorTest.kt [hunk 1]
```diff
@@ -63,18 +58,18 @@ class PostLoginTest : StringSpec() {
...
-                assertionConsumerServiceURL = acsUrl[HTTP_POST]
-                id = TestCommon.ID
+                assertionConsumerServiceURL = acsUrl[SamlProtocol.Binding.HTTP_POST]
+                id = ID
                 version = SAMLVersion.VERSION_20
                 issueInstant = DateTime()
-                destination = Common.getSingleSignOnLocation(POST_BINDING)
-                protocolBinding = POST_BINDING
+                destination = Common.getSingleSignOnLocation(SamlProtocol.POST_BINDING)
+                protocolBinding = SamlProtocol.POST_BINDING
```
Comment (blen-desta) [inline ctk/idp/src/main/kotlin/org/codice/compliance/web/sso/error/PostSSOErrorTest.kt]: ✏️ You could import `org.codice.security.saml.SamlProtocol.POST_BINDING` and `org.codice.security.saml.SamlProtocol.Binding.HTTP_POST` to make it cleaner in the test like we have it in the positive tests

File: deployment/suites/src/main/kotlin/org/codice/compliance/tests/suites/BasicTestsSuite.kt [hunk 2]
```diff
@@ -13,12 +13,16 @@
...
+import org.codice.compliance.web.sso.RedirectSSOTest
+import org.codice.compliance.web.sso.error.PostSSOErrorTest
+import org.codice.compliance.web.sso.error.RedirectSSOErrorTest
 import org.junit.runner.RunWith
 import org.junit.runners.Suite
 
 @RunWith(Suite::class)
-@Suite.SuiteClasses(PostLoginTest::class,
-        RedirectLoginTest::class)
+@Suite.SuiteClasses(PostSSOTest::class,
```
Comment (blen-desta) [inline deployment/suites/src/main/kotlin/org/codice/compliance/tests/suites/BasicTestsSuite.kt]: ❓ Should we remove the negative tests from the test suite?

File: ctk/idp/src/main/kotlin/org/codice/compliance/web/sso/PostSSOTest.kt [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+import org.opensaml.saml.saml2.core.impl.IssuerBuilder
+import org.opensaml.saml.saml2.core.impl.NameIDPolicyBuilder
+
+class PostSSOTest : StringSpec() {
+    companion object {
+
+        /** Sets up positive path tests.
+         * @return A string representation of a valid encoded POST AuthnRequest.
+         */
+        private fun createValidAuthnRequest(): String {
```
Comment (blen-desta) [inline ctk/idp/src/main/kotlin/org/codice/compliance/web/sso/PostSSOTest.kt]: ❓ Should we extract `createValidAuthnRequest` to `TestCommon`? One for Post and one for Redirect

File: deployment/suites/src/main/kotlin/org/codice/compliance/tests/suites/BasicTestsSuite.kt (hunk 2 above)
Comment (coyotesqrl) [reply deployment/suites/src/main/kotlin/org/codice/compliance/tests/suites/BasicTestsSuite.kt]: Maybe we should have a positive and a negative suite?

Comment (brjeter) [reply deployment/suites/src/main/kotlin/org/codice/compliance/tests/suites/BasicTestsSuite.kt]: I would remove them from the suite, but once we fix relay state in DDF we're going to turn the negative relay state test back on right? There's another "negative" test that doesn't return a SAML error that we might turn on as well.

Comment (blen-desta) [reply deployment/suites/src/main/kotlin/org/codice/compliance/tests/suites/BasicTestsSuite.kt]: We can separate them and still run the negative tests suite. At that point we will enable that test. We can also add a way to turn it off. Maybe by passing in a flag when we run Keycloak

File: ctk/idp/src/main/kotlin/org/codice/compliance/web/sso/PostSSOTest.kt (hunk 3 above)
Comment (brjeter) [reply ctk/idp/src/main/kotlin/org/codice/compliance/web/sso/PostSSOTest.kt]: I will do this in a follow-on PR where I have to rework some of these utility functions anyways.

**Comment by brjeter** [comment]
build now

**Pull request merged**